src/http-gateways/path-gateway.md

```diff
@@ -205,6 +205,30 @@ This is a URL-friendly alternative to sending an [`Accept`](#accept-request-head
 - `format=json` → `Accept: application/json`
 - `format=cbor` → `Accept: application/cbor`
 
+## Query Parameters for CAR Requests
+
+The following query parameters are only available for requests made with either a `format=car` query parameter or an `Accept: application/vnd.ipld.car` request header. These parameters modify shape of the IPLD graph returned within the car file
+
+### `car-scope` (request query parameter)
+
+Optional, `car-scope=(block|file|all)` with default value 'all', describes the shape of the dag fetched the terminus of the specified path whose blocks are included in the returned CAR file after the blocks required to traverse path segments.
+
+`block` - Only the root block at the end of the path is returned After blocks required to verify the specified path segments.
+
+`file` - For queries that traverse UnixFS data, `file` roughly means return blocks needed to verify the end of the path as a filesystem entity. In other words, all the blocks needed 'cat' a UnixFS file at the end of the specified path, or to 'ls' a UnixFS directory at the end of the specified path. For all queries that do not reference non-UnixFS data, `file` is equivalent to `block`
+
+`all` - Transmit the entire contiguous DAG that begins at the end of the path query, after blocks required to verify path segments
```

Comment:
Good question, if we're first-classing unixfs in all of this then it's not unreasonable to suggest we should interpret "all" via a unixfs lens. This would require more complexity though, we don't have enough pieces implemented via go-unixfsnode to make this workable just yet I think.

Another variation might be a unixfs link pointing to a non-unixfs graph, should that be followed, result in an error, be ignored?

Reply:
> we don't have enough pieces implemented via go-unixfsnode to make this workable just yet I think.

I can't speak to that, although it seems a bit weird if true given that we can do this with what was formerly go-unixfs. I suspect it's probably doable with go-unixfsnode but if anything the difficulty would arise in expressing this via selectors which I think might be doable as well.

> Another variation might be a unixfs link pointing to a non-unixfs graph, should that be followed, result in an error, be ignored?

Yeah, that's perhaps an even better example. Mostly we should clarify what's going on here. We might be able to punt on what the "ideal" model is here since IIUC for non-malformed UnixFS "all as IPLD" or "all as UnixFS" are the same. However, this probably means more pressure on tools like https://github.com/ipfs/gateway-conformance to make sure we handle malformed UnixFS uniformly by erroring.

My understanding (from watching a several hundred message long Slack thread that has long since been GC'd because people didn't use public forums 😢) is that not erroring on undefined edge cases is the reason why you can do `/ipfs/some-dag-cbor-thing/path/to/unixfs-dir/file`. So I'd like to keep these cases covered so that we don't get spec expansion by accident (e.g. people trying to hookup non-unixfs graphs to unixfs ones and asking gateways to traverse them).

Reply:
ACK on the ADL thing, we shouldn't drop that line of thinking and you're right that something in the region of @rangermauve's experiments are probably the path forward here. I'd like to mature one of my side experiments to the point where a stacked ADL makes sense so we can push on this a bit.

Reply:
I've pushed IPIP with some examples of test scenarios.

Gateways support traversing DAG-CBOR  like `/ipfs/some-dag-cbor-thing/path/to/unixfs-dir/file`.

> Does this intentionally use the underlying IPLD set of links rather than logical data?

Yes,  for the meaning of `all`, I've added a note about following all IPLD Links recursively for the sake of simplicity -- lmk if that is good enough.